source/extensions/common/wasm/context.cc

```diff
@@ -1125,13 +1132,30 @@ WasmResult Context::setProperty(absl::string_view path, absl::string_view value)
   if (!stream_info) {
     return WasmResult::NotFound;
   }
+  
   std::string key;
+  auto part_token = property_tokens.find(path);
+  
+  if (part_token->second == PropertyToken::DYNAMICMETADATA) {
+    ProtobufWkt::Struct metadata_val;
+    auto& fields_a = *metadata_val.mutable_fields();
+    std::string v1;
+    absl::StrAppend(&v1, value);
+    fields_a[v1].set_bool_value(true);
+    std::string key3;
+    absl::StrAppend(&key3, path);
+    stream_info->setDynamicMetadata(key3, metadata_val);
```

Comment:
Unclear how key and value are derived.

Reply:
JSON sounds fine, although it's parsing is pretty slow. Using dynamic metadata has inherent performance issues, so it might be ok with parsing. WDYT @PiotrSikora ?

Reply:
We changed the PR, the idea is to parse the path, identify the metadata and set the dynamic metadata.
client side would be:
```rust
   self.set_property(vec!["metadata","my.filters.network.network"], Some("{\"noaccess\": true}".as_ref()));
```
cc @kyessenov

Reply:
Note that using JSON to `set_property` like the above for metadata does not match the intuition of getting back JSON from `get_property` with metadata - in particular, `get_property` can get you a protobuf struct when asking just for `vec!["metadata"]` or or a "pairs" length and 0-delimited key-value array without type information for values if you specify a path after "metadata" (at least for the use cases I've had).

I think there is some work to be done in this area to define what is expected in both directions.